sei-db/sc/memiavl/db.go

```diff
@@ -592,10 +667,30 @@ func (db *DB) rewriteIfApplicable(height int64) {
 		return
 	}
 
-	// create snapshot when current height - last snapshot height > interval
-	if height-db.SnapshotVersion() >= int64(db.snapshotInterval) {
-		// Try prune old snapshots before starting new snapshot
-		db.pruneSnapshots()
+	snapshotVersion := db.SnapshotVersion()
+	blocksSinceLastSnapshot := height - snapshotVersion
+
+	// Create snapshot when both conditions are met:
+	// 1. Block height interval is reached (height - last snapshot >= interval)
+	// 2. Minimum time interval has elapsed (prevents excessive snapshots during catch-up)
+	if blocksSinceLastSnapshot >= int64(db.snapshotInterval) {
+		timeSinceLastSnapshot := time.Since(db.lastSnapshotTime)
+		minTimeInterval := time.Duration(db.snapshotMinTimeInterval) * time.Second
```

Comment:
Looks like we are converting things to drutation. Would be good to change the config at leat at the level of DB struct instantiation to take `time.Duration` straight-up instead of `int32`.